### ISSUE TITLE
Where do the console logs go?

### ISSUE THREAD

**Issue by mark-hallman** [EricZimmerman/KapeFiles#137]
Riddle me this Batman  **Where do the console logs go in this scenario?** 

 A computer (Lab-02) executes kape.exe from a UNC path `\\Lab03\c$\\Tools\KAPE\kape.exe`

`The folder \\Lab03\c$\Tools\KAPE\` contains a "_kape.cli" batch file that directs KAPE's actions.

The target and module destinations used in the batch file are on a different UNC path `--tdest \\Lab-04\c$\kape_out\tdest`

**Where do the console logs go?**  I can't find the console logs on any of the machines and I'm not sure where to even expect them to go other than destination dirs.   Having the console logs would be helpful in debugging this.  I know that the command gets executed (maybe not successfully) because the `_kape.cli` gets renamed on Lab-03. The output never gets to the destination on LAB-04. Therefore, I can't look for the console logs for help with this.   If I run the exact same command directly from Lab-02, still using the `kape.exe` on the same UNC path, it works perfectly. [see command below]

Here is the one line "_kape.cli"  batch file that reside in \\Lab03\c$\Tools\KAPE\`

```--tsource C --target LnkFilesAndJumpLists --tdest \\Lab-04\c$\kape_out\tdest --module LECmd,JLECmd --mdest \\Lab-04\c$\kape_out\mdest --debug --trace```

This is the command that runs correctly when run from from Lab-02.
```\\lab-03\c$\Tools\KAPE\kape.exe --tsource C --target LnkFilesAndJumpLists --tdest \\Lab-04\c$\kape_out\tdest --module LECmd,JLECmd --mdest \\Lab-04\c$\kape_out\mdest```

Is where a way direct where the console log goes?  Does it just get dropped since the destination(s) writes failed? I have noticed in a lot of testing of my own KAPE scripts that a command will fail and  I just get the KAPE usage back and no message of any kind on why it failed.  KAPE usage =KAPE help you see when you just type kape.exe on the command line. Some kind of message would be nice. 

**Thanks for all the time that you put into these tools for the rest of us to use.**

**Comment by EricZimmerman** [comment]
console logs should end up in --tdest and/or --mdest

if both are set, you get copies in both

you can make your own copy of console log via > or whatever method your shell provides.

if destination write fails you have an issue. the screen should shjow this.

kape generally tries to write the consolelog to its own directory if an unhandled exception occurs

what do you see on the screen while its running, or cant you?

**Comment by EricZimmerman** [comment]
my suggestion would be to get things running via a single command line you can watch first, then automate when you know settings are right

**Comment by mark-hallman** [comment]
Eric,

I have tried the redirect and using `--gui` and not using` --gui` but I don't even get any popup or output to the screen when I start the process from Lab-01.  If I move over to Lab-02 and run the same command, still using the remote exe and a batch file it runs and I do get the console log.

I have a test script that basically does a `dir` to test that the creds are working to make it through the double hop. A runs a command on B that then runs a command on C.   So,  I run a dir command on A that give me a dir on C.  

```Invoke-Command -ComputerName LAB-02 -ScriptBlock {Get-ChildItem -Path \\Lab-03\c$ } -ConfigurationName KAPE```

Agreed on the process of getting a basic command running and working progressively toward the desired command.  

Closing for now.  I'll try the reredirect even though I not seeing anything.
